hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/NodeStateManager.java

```diff
@@ -501,6 +518,26 @@ public List<DatanodeInfo> getNodes(
     return nodeStateMap.getDatanodeInfos(opState, health);
   }
 
+  public List<DatanodeInfo> getVolumeFailuresNodes() {
+    List<DatanodeInfo> allNodes = nodeStateMap.getAllDatanodeInfos();
+    if (allNodes.size() < 1) {
+      return allNodes;
+    }
+
+    List<DatanodeInfo> failedVolumeNodes = new ArrayList<>();
+    for (DatanodeInfo dn : allNodes) {
+      if (dn.getFailedVolumeCount() > 0) {
+        failedVolumeNodes.add(dn);
+      }
+    }
+
+    return failedVolumeNodes;
+  }
+
+  public int getVolumeFailuresNodeCount() {
```

Comment:
Javadoc

Reply:
Thanks @ivandika3  for the comment and review.
I have updated it.